burn-tensor/src/tensor/ops/tensor.rs

```diff
@@ -137,11 +137,36 @@ pub trait TensorOps<B: Backend> {
     ///
     /// The tensor with the given values.
     fn arange(range: Range<usize>, device: &B::Device) -> B::IntTensorPrimitive<1> {
-        let shape = Shape::new([range.end - range.start]);
         let value = range
             .into_iter()
             .map(|i| (i as i64).elem())
             .collect::<Vec<B::IntElem>>();
+        let shape = Shape::new([value.len()]);
+        let data = Data::new(value, shape);
+        B::int_from_data(data, device)
+    }
```

Comment:
The default implementation should call `arange_step` instead. Most backends could only implement `arange_step` for performance and totaly ignore `arange`.